### ISSUE TITLE
Update Contributing.md Guide

### ISSUE THREAD

**Issue by dawoodkhan82** [gradio-app/gradio#3247]
Provide more details and steps for setting up local dev environment and start running/debugging a demo.

**Comment by abidlabs** [comment]
It seems a lot of people also get stuck when using Codespaces so it'd be nice to make sure that DX flow is streamlined. Related issue: #2343

**Comment by abidlabs** [comment]
A few other things I noticed @dawoodkhan82:
- we no longer use CircleCI so this should be removed from contributing.md
- we have added a linter but this is not mentioned in contributing.md

Also the PR template is a little bloated, perhaps we can trim that a bit as well

**Comment by thepelkus-too** [comment]
A note on testing in that document:

Under [Continuous Integration and Testing](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md#continuous-integration-and-testing) it says:
> All PRs must pass the continuous integration tests before merging. To test locally, you can run python -m unittest from the repo directory.

When I ran that command, no tests were run. However, when I ran the test command from the [Extra tidbits](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md#extra-tidbits):
> bash scripts/run_all_tests.sh

it seemed like it might've been doing what I would expect.

I had some difficulties with those tests that might've stemmed from running in the Windows Subsystem for Linux, but I'm not sure just yet; I'll make another note here if I can pin that down.
